From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shane Freeder <theboyetronic@gmail.com>
Date: Tue, 24 Dec 2019 00:35:42 +0000
Subject: [PATCH] PlayerDeathEvent#shouldDropExperience


diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index cf75fcbf90d481f95eb85644b7931ebc50922f8e..cdfcf0e819d434f8c49d5fb5fe0a1f38615bcf59 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -818,7 +818,7 @@ public class ServerPlayer extends Player implements ContainerListener {
             this.tellNeutralMobsThatIDied();
         }
         // SPIGOT-5478 must be called manually now
-        this.dropExperience();
+        if (event.shouldDropExperience()) this.dropExperience(); // Paper - tie to event
         // we clean the player's inventory after the EntityDeathEvent is called so plugins can get the exact state of the inventory.
         if (!event.getKeepInventory()) {
             // Paper start - replace logic
