From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Blake Galbreath <blake.galbreath@gmail.com>
Date: Sat, 20 Apr 2019 19:47:34 -0500
Subject: [PATCH] Expose the internal current tick


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 4906b8e70f063e4848be84bde96ec762d09ee78f..c5a111af15abceda9a87a794ec181bd4abc43c6a 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2641,5 +2641,10 @@ public final class CraftServer implements Server {
         profile.getProperties().putAll(((CraftPlayer)player).getHandle().getGameProfile().getProperties());
         return new com.destroystokyo.paper.profile.CraftPlayerProfile(profile);
     }
+
+    @Override
+    public int getCurrentTick() {
+        return net.minecraft.server.MinecraftServer.currentTick;
+    }
     // Paper end
 }
